src/frontend/komponenter/Fagsak/Vilkårsvurdering/GeneriskVilkår/GeneriskVilkårVurdering.tsx

```diff
@@ -89,12 +96,58 @@ const GeneriskVilkårVurdering: React.FC<IProps> = ({
     const onClickVilkårFerdig = () => {
         const validertVilkår = redigerbartVilkår.valideringsFunksjon(redigerbartVilkår, person);
 
-        settVilkårForPeriodeResultat(person.personIdent, redigerbartVilkår);
-        if (validertVilkår.valideringsstatus === Valideringsstatus.OK) {
-            settEkspandertVilkår(false);
-            settVisFeilmeldingerForEttVilkår(false);
+        const vilkårsvurderingForPerson = vilkårsvurdering.find(
+            (personResultat: IPersonResultat) => personResultat.personIdent === person.personIdent
+        );
+
+        lagreVilkår(validertVilkår, vilkårsvurderingForPerson);
+    };
+
+    const lagreVilkår = (
+        validertVilkår: IFelt<IVilkårResultat>,
+        vilkårsvurderingForPerson: IPersonResultat | undefined
+    ) => {
+        if (
+            vilkårsvurderingForPerson &&
+            validertVilkår.valideringsstatus === Valideringsstatus.OK
+        ) {
+            putVilkår(vilkårsvurderingForPerson, redigerbartVilkår)
+                .then((nyVilkårsvurdering: Ressurs<IRestPersonResultat[]>) => {
+                    settLagrerVilkår(false);
+                    if (nyVilkårsvurdering.status === RessursStatus.SUKSESS) {
+                        settVilkårsvurderingFraApi(nyVilkårsvurdering.data);
+                        settEkspandertVilkår(false);
+                        settVisFeilmeldingerForEttVilkår(false);
+                    } else if (nyVilkårsvurdering.status === RessursStatus.FEILET) {
+                        settVisFeilmeldingerForEttVilkår(true);
+                        settRedigerbartVilkår({
+                            ...redigerbartVilkår,
+                            valideringsstatus: Valideringsstatus.FEIL,
+                            feilmelding: nyVilkårsvurdering.frontendFeilmelding,
+                        });
+                    } else {
```

Comment:
Er det meningen av kun `RessursStatus.SUKSESS` og catchen nede har `settLagrerVilkår(false);` ? Regner med dette kanskje gjør noe med spinneren og sånn? I så fall bør det kanskje gjøres i hver eller helt på slutten?

Reply:
Denne else-en er gjør da også det samme som catchen og man kan kanskje bare kaste en feil i else? :)

Reply:
Ah, dette er kanskje ikke så lesbart? `settLagrerVilkår` er default `false`. Inne i `putVilkår` blir den satt til `true`. Når den returneres skal den settes tilbake til `false` uavhengig av svaret fra `putVilkår`. `settLagrerVilkår`ligger utenfor if-else-utrykket og skjer for så vidt ikke bare på suksess, eller misforstår jeg?

Reply:
Good catch med den else-en. Den kan definitivt bare kaste en feil :-)

Reply:
else-en tror jeg vi er avhengig av fordi vi er nødt til å gjøre de operasjonene hvis den ytterste if-en feiler. Catch-en vil ikke fange det opp da.